core/src/main/java/com/bankbazaar/core/manager/TripUserMapEntityManager.java

```diff
@@ -0,0 +1,101 @@
+package com.bankbazaar.core.manager;
+
+import com.bankbazaar.core.model.TripEntity;
+import com.bankbazaar.core.model.TripUserMapEntity;
+import com.bankbazaar.core.model.UserEntity;
+import com.bankbazaar.core.repository.TripRepository;
+import com.bankbazaar.core.repository.TripUserMapRepository;
+import com.bankbazaar.core.model.TripUserCompositeKey;
+import com.bankbazaar.core.repository.UserRepository;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+@Slf4j
+@Service
+public class TripUserMapEntityManager {
+
+    @Autowired
+    private TripUserMapRepository tripUserMapRepository;
+    @Autowired
+    private TripRepository tripRepository;
+    @Autowired
+    private UserRepository usersRepository;
+
+    /**
+     * Insert to TripUserMapping table
+     * @param data
+     */
+    public TripUserMapEntity saveTripUserMapping(TripUserMapEntity data)
+    {
+        Optional<TripEntity> tripId = tripRepository.findById(data.getTripId().getTripId());
+        Optional<UserEntity> userId = usersRepository.findById(data.getUserId().getUserId());
+        data.setTripId(tripId.get());
```

Comment:
It says tripId and userId but we are setting the entire objects

Reply:
Same like the problem above. after setting columns to nullable=false, i have to update "tripId" and "userId" foreign key objects with the existing data .